### ISSUE TITLE
RuntimeError: maximum recursion depth exceeded while calling a Python object

### ISSUE THREAD

**Issue by GitterRemote** [ets-labs/python-dependency-injector#245]
```python
    D = collections.namedtuple("D", ['e'])                                       
    E = collections.namedtuple("E", ['provide_d'])                               
                                                                                 
    class X(containers.DeclarativeContainer):                                    
        e = providers.Singleton(E)                                               
        d = providers.Singleton(D, e=e)                                          
        e.add_args(d.delegate())                                                 
                                                                                 
    assert X.d()                                                                 
    assert X.d().e                                                               
    assert X.d().e.provide_d()                                                  
    assert X.d().e.provide_d() is X.d()                                          
    X()  # raise RuntimeError: maximum recursion depth exceeded while calling a Python object
```

**Comment by rmk135** [comment]
Thanks, I'll see if the recursion can be avoided in that case.

**Comment by rmk135** [comment]
I fixed this in `4.29.1`. I'm sorry it took almost a year to get there. Later is better than never.
